### PR TITLE
Fix Rawhide FTBFS

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Fix Rawhide FTBFS. (Fixes: #1743, Fixes: #1746)

### CODE REVIEW

File: src/server/frontend_xwayland/xwayland_surface.cpp [hunk 1]
```diff
@@ -137,7 +137,7 @@ auto property_handler(
     xcb_atom_t property,
     std::function<void(T const&)> handler) -> std::pair<xcb_atom_t, std::function<std::function<void()>()>>
 {
-    return property_handler<T>(connection, window, property, {handler});
+    return property_handler<T>(connection, window, property, mf::XCBConnection::Handler<T>{std::move(handler)});
```
Comment (Saviq) [inline src/server/frontend_xwayland/xwayland_surface.cpp:140]: This doesn't seem to help with a segfault I'm getting on the performance tests on the Pi:
```
0x0000fffff7db3f04 in std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>::function(std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&) (
    this=0xffffd8ffb048, __x=...) at /usr/include/c++/9/bits/std_function.h:263
263         bool _M_empty() const { return !_M_manager; }
(gdb) bt
#0  0x0000fffff7db3f04 in std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>::function(std::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)> const&)
    (this=0xffffd8ffb048, __x=...) at /usr/include/c++/9/bits/std_function.h:263
#1  0x0000fffff7dbe8e0 in (anonymous namespace)::property_handler<std::__cxx11::basic_string<char> >(const std::shared_ptr<mir::frontend::XCBConnection> &, xcb_window_t, xcb_atom_t, std::function<void(const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)>) (connection=
    std::shared_ptr<mir::frontend::XCBConnection> (use count 4, weak count 0) = {...}, window=window@entry=4194306, property=property@entry=67, handler=...) at ./src/server/frontend_xwayland/xwayland_surface.cpp:134
#2  0x0000fffff7dbe8f8 in (anonymous namespace)::property_handler<std::__cxx11::basic_string<char> >(const std::shared_ptr<mir::frontend::XCBConnection> &, xcb_window_t, xcb_atom_t, std::function<void(const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)>)
    (connection=std::shared_ptr<mir::frontend::XCBConnection> (use count 4, weak count 0) = {...}, window=window@entry=4194306, property=property@entry=67, handler=...) at ./src/server/frontend_xwayland/xwayland_surface.cpp:134
# and so on
```

**Comment by AlanGriffiths** [comment]
Are you sure you have the 2nd commit?

On Tue, 29 Sep 2020 10:26:36 -0700, Michał Sawicz wrote:
> @Saviq commented on this pull request.
>
>
>
>> -    return property_handler<T>(connection, window, property, {handler});
> +    return property_handler<T>(connection, window, property, 
> mf::XCBConnection::Handler<T>{std::move(handler)});
>
> This doesn't seem to help with a segfault I'm getting on the 
> performance tests on the Pi:
> ```gdb
> 0x0000fffff7db3f04 in std::function<void 
> (std::__cxx11::basic_string<char, std::char_traits<char>, 
> std::allocator<char> > const&)>::function(std::function<void 
> (std::__cxx11::basic_string<char, std::char_traits<char>, 
> std::allocator<char> > const&)> const&) (
>     this=0xffffd8ffb048, __x=...) at 
> /usr/include/c++/9/bits/std_function.h:263
> 263         bool _M_empty() const { return !_M_manager; }
> (gdb) bt
> #0  0x0000fffff7db3f04 in std::function<void 
> (std::__cxx11::basic_string<char, std::char_traits<char>, 
> std::allocator<char> > const&)>::function(std::function<void 
> (std::__cxx11::basic_string<char, std::char_traits<char>, 
> std::allocator<char> > const&)> const&)
>     (this=0xffffd8ffb048, __x=...) at 
> /usr/include/c++/9/bits/std_function.h:263
> #1  0x0000fffff7dbe8e0 in (anonymous 
> namespace)::property_handler<std::__cxx11::basic_string<char> 
> >(const std::shared_ptr<mir::frontend::XCBConnection> &, 
> xcb_window_t, xcb_atom_t, std::function<void(const 
> std::__cxx11::basic_string<char, std::char_traits<char>, 
> std::allocator<char> >&)>) (connection=
>     std::shared_ptr<mir::frontend::XCBConnection> (use count 4, 
> weak count 0) = {...}, window=window@entry=4194306, 
> property=property@entry=67, handler=...) at 
> ./src/server/frontend_xwayland/xwayland_surface.cpp:134
> #2  0x0000fffff7dbe8f8 in (anonymous 
> namespace)::property_handler<std::__cxx11::basic_string<char> 
> >(const std::shared_ptr<mir::frontend::XCBConnection> &, 
> xcb_window_t, xcb_atom_t, std::function<void(const 
> std::__cxx11::basic_string<char, std::char_traits<char>, 
> std::allocator<char> >&)>)
>     (connection=std::shared_ptr<mir::frontend::XCBConnection> 
> (use count 4, weak count 0) = {...}, 
> window=window@entry=4194306, property=property@entry=67, 
> handler=...) at 
> ./src/server/frontend_xwayland/xwayland_surface.cpp:134
> # and so on
> ```
>

-- 
Sent using Dekko from my Ubuntu device

**Comment by Saviq** [comment]
> Are you sure you have the 2nd commit?

Indeed I did not. Had a mix of built and installed pieces of mir around.

I can confirm this fixes #1746.

**Review by Saviq — APPROVED** [review]

**Comment by Saviq** [comment]
Let's.

bors merge

**Pull request merged**